docker-cargo.sh

```diff
@@ -3,72 +3,84 @@ REPO_ROOT=$(readlink -f $(dirname $(readlink -f $0)))
 OUTPUT_DIR=docker-cargo/
 CARGO_HOME=${CARGO_HOME:-$HOME/.cargo}
 
-
-DOCKER_BUILDER_IMAGE=${DOCKER_BUILDER_IMAGE:-mangatasolutions/node-builder:0.2}
+DOCKER_BUILDER_IMAGE=${DOCKER_BUILDER_IMAGE:-mangatasolutions/node-builder:multi}
 DOCKER_USER="$(id -u):$(id -g)"
 DOCKER_JOB_NAME=cargo-wrapper
 if [ -n "${DISABLE_TTY}" ]; then
-    ALLOCATE_TTY_OR_NOT="-i"
+  ALLOCATE_TTY_OR_NOT="-i"
 else
-    ALLOCATE_TTY_OR_NOT="-it"
+  ALLOCATE_TTY_OR_NOT="-it"
 fi
 
 CARGO_COMMAND=$1
 
 CARGO_ARGS=${@:2}
 
 if [ "$CARGO_COMMAND" == "kill" ]; then
-    docker kill ${DOCKER_JOB_NAME}
-    exit 0
+  docker kill ${DOCKER_JOB_NAME}
+  exit 0
 fi
 
-if ! which docker > /dev/null; then
-	echo "docker not installed" >&2
-	exit -1
+if ! which docker >/dev/null; then
+  echo "docker not installed" >&2
+  exit 1
 fi
 
-if docker inspect ${DOCKER_BUILDER_IMAGE} > /dev/null; then
-	echo "building using docker image ${DOCKER_BUILDER_IMAGE}"
+if docker inspect ${DOCKER_BUILDER_IMAGE} >/dev/null; then
+  echo "building using docker image ${DOCKER_BUILDER_IMAGE}"
 else
-	echo "docker image ${DOCKER_BUILDER_IMAGE} not found - pulling" >&2
-	docker pull ${DOCKER_BUILDER_IMAGE}
+  echo "docker image ${DOCKER_BUILDER_IMAGE} not found - pulling" >&2
+  docker pull ${DOCKER_BUILDER_IMAGE}
 fi
 
-if [ -e ${CARGO_HOME} ] ; then
+if [ -n "${REUSE_LOCAL_CACHE}" ]; then
```

Comment:
by default, the caches will be store in OUTPUT_DIR
to override set REUSE_LOCAL_CACHE=1